src/rgw/rgw_rados.h

```diff
@@ -1707,6 +1711,116 @@ class RGWRados
   static uint32_t calc_ordered_bucket_list_per_shard(uint32_t num_entries,
 						     uint32_t num_shards);
 };
+
+
+// ANCHOR: #CACHEREBASE
+
+struct get_obj_aio_data {
+  struct get_obj_data *op_data;
+  off_t ofs;
+  off_t len;
+};
+
+struct get_obj_io {
+  off_t len;
+  bufferlist bl;
+};
+
+class librados::CacheRequest {
+  public:
+    ceph::mutex lock;
+    int sequence;
+    bufferlist *pbl;
+    struct get_obj_data *op_data;
+    std::string oid;
+    off_t ofs;
+    off_t len;
+    librados::AioCompletion *lc;
+    std::string key;
+    off_t read_ofs;
+    Context *onack;
+    CephContext *cct;
+    CacheRequest(CephContext *_cct) : lock("CacheRequest"), sequence(0), pbl(NULL), op_data(NULL), ofs(0), lc(NULL), read_ofs(0), cct(_cct) {};
+    virtual ~CacheRequest(){};
+    virtual void release()=0;
+    virtual void cancel_io()=0;
+    virtual int status()=0;
+    virtual void finish()=0;
+};
+
+struct librados::L1CacheRequest : public librados::CacheRequest{
+  int stat;
+  struct aiocb *paiocb;
+  L1CacheRequest(CephContext *_cct) :  CacheRequest(_cct), stat(-1), paiocb(NULL) {}
+  ~L1CacheRequest(){}
+  void release (){
+    lock.lock();
+    free((void *)paiocb->aio_buf);
```

Comment:
you never need to cast to a void*

Reply:
seems like that this is a corner case, without the casting there is an error:
`argument of type "volatile void *" is incompatible with parameter of type "void *"`

`aio_buf` is defined as volatile void pointer at: `/usr/include/aio.h`
`volatile void *aio_buf;	/* Location of buffer.  */`